### PR TITLE
Adds founder rank

### PR DESCRIPTION
**Pull request by ghost**
Adds founder rank
Removes Wild from dev list and updates dev list with newest devs.
Adds RainbowChat Array
Added markbyron to Founder Array.
made Impostors gets editmuted

### CODE REVIEW

File: src/main/java/me/totalfreedom/totalfreedommod/FrontDoor.java
Comment (Wild1145) [inline src/main/java/me/totalfreedom/totalfreedommod/FrontDoor.java]: Formatting here please.

File: src/main/java/me/totalfreedom/totalfreedommod/rank/RankManager.java
Comment (Wild1145) [inline src/main/java/me/totalfreedom/totalfreedommod/rank/RankManager.java:167]: Why have you done this? I can see no logic to this.

File: src/main/java/me/totalfreedom/totalfreedommod/rank/Title.java
Comment (Wild1145) [inline src/main/java/me/totalfreedom/totalfreedommod/rank/Title.java]: This should be "Server Founder" as this plugin does also get used on other servers, or even just "The Founder"

File: src/main/java/me/totalfreedom/totalfreedommod/util/FUtil.java
Comment (Wild1145) [inline src/main/java/me/totalfreedom/totalfreedommod/util/FUtil.java]: Why you have removed me I do not know, this is a list of TFM developers, not TF developers. You are also missing the comment above I see that does say names should not be removed.

File: src/main/java/me/totalfreedom/totalfreedommod/util/FUtil.java
Comment (Wild1145) [inline src/main/java/me/totalfreedom/totalfreedommod/util/FUtil.java]: This should be handled as a config entry

File: src/main/java/me/totalfreedom/totalfreedommod/util/FUtil.java
Comment (Wild1145) [inline src/main/java/me/totalfreedom/totalfreedommod/util/FUtil.java]: Why is this here?

**Review by Wild1145 — CHANGES_REQUESTED** [review]
A few things to look at, usually I would say to split the colour rainbow stuff from the founder stuff but given we're being a bit more flexible right now, I wont worry. In the future though, please keep differing functionality to their own PR's, that way things get through the process quicker.

File: src/main/java/me/totalfreedom/totalfreedommod/rank/RankManager.java
Comment (Wild1145) [inline src/main/java/me/totalfreedom/totalfreedommod/rank/RankManager.java]: Why have you done this?

File: src/main/java/me/totalfreedom/totalfreedommod/rank/RankManager.java
Comment (ghost) [reply src/main/java/me/totalfreedom/totalfreedommod/rank/RankManager.java:167]: To make Founder rank if is supered and mark uses a custom login msg , it can be replacen with mark's custom loginmsg.

File: src/main/java/me/totalfreedom/totalfreedommod/util/FUtil.java
Comment (Wild1145) [inline src/main/java/me/totalfreedom/totalfreedommod/util/FUtil.java:60]: Can we not use the above pool of colours?

File: src/main/java/me/totalfreedom/totalfreedommod/rank/Title.java
Comment (ghost) [reply src/main/java/me/totalfreedom/totalfreedommod/rank/Title.java]: Mark used that tag and asked me to set like his tag.

**Review by Wild1145 — CHANGES_REQUESTED** [review]

File: src/main/java/me/totalfreedom/totalfreedommod/util/FUtil.java
Comment (ghost) [reply src/main/java/me/totalfreedom/totalfreedommod/util/FUtil.java]: markbyron asked me to remove your name from developers.

Comment (Wild1145) [reply src/main/java/me/totalfreedom/totalfreedommod/util/FUtil.java]: That makes no difference. This is a list of TFM developers, and as per the license names are not to be removed from this list.

File: src/main/java/me/totalfreedom/totalfreedommod/util/FUtil.java
Comment (ghost) [reply src/main/java/me/totalfreedom/totalfreedommod/util/FUtil.java]: uh, maybe madgeeks's edits? i didnt notice that

File: src/main/java/me/totalfreedom/totalfreedommod/util/FUtil.java
Comment (ghost) [reply src/main/java/me/totalfreedom/totalfreedommod/util/FUtil.java]: markbyron is the founder , without him TF wont exist and the founder it should be hard-coded.

File: src/main/java/me/totalfreedom/totalfreedommod/rank/RankManager.java
Comment (ghost) [reply src/main/java/me/totalfreedom/totalfreedommod/rank/RankManager.java]: Because the freeze doesn't work.

File: src/main/java/me/totalfreedom/totalfreedommod/util/FUtil.java
Comment (ghost) [reply src/main/java/me/totalfreedom/totalfreedommod/util/FUtil.java:60]: is using from a current command (an alternative to nicknyan)

File: src/main/java/me/totalfreedom/totalfreedommod/rank/RankManager.java
Comment (Wild1145) [reply src/main/java/me/totalfreedom/totalfreedommod/rank/RankManager.java:167]: @marcocorriero How is the owner login message currently handled then?

File: src/main/java/me/totalfreedom/totalfreedommod/rank/Title.java
Comment (Wild1145) [reply src/main/java/me/totalfreedom/totalfreedommod/rank/Title.java]: It would make it inaccurate on other servers using TFM. It should be generic to be used on all servers, or a system put in place to check if this plugin is running on TF or not, and only display that rank on TF.

File: src/main/java/me/totalfreedom/totalfreedommod/util/FUtil.java
Comment (ghost) [reply src/main/java/me/totalfreedom/totalfreedommod/util/FUtil.java]: fine.
adding you back

File: src/main/java/me/totalfreedom/totalfreedommod/util/FUtil.java
Comment (Wild1145) [reply src/main/java/me/totalfreedom/totalfreedommod/util/FUtil.java]: Can you remove if from your PR then please? It just doenst appear to have any functionality.

File: src/main/java/me/totalfreedom/totalfreedommod/rank/RankManager.java
Comment (Wild1145) [reply src/main/java/me/totalfreedom/totalfreedommod/rank/RankManager.java]: It still doesnt explain why you've changed it?

File: src/main/java/me/totalfreedom/totalfreedommod/util/FUtil.java
Comment (Wild1145) [reply src/main/java/me/totalfreedom/totalfreedommod/util/FUtil.java:60]: I mean why add another pool of colours when we could merge the two pools into one?

File: src/main/java/me/totalfreedom/totalfreedommod/util/FUtil.java
Comment (Wild1145) [reply src/main/java/me/totalfreedom/totalfreedommod/util/FUtil.java]: Again, makes no difference. It should be configured in config like the owner is.

File: src/main/java/me/totalfreedom/totalfreedommod/util/FUtil.java
Comment (ghost) [reply src/main/java/me/totalfreedom/totalfreedommod/util/FUtil.java:60]: because look at the color, is different, and this pool uses a specific order (1, 2, 3 4) thanks to the iterator. not a misc order.

File: src/main/java/me/totalfreedom/totalfreedommod/util/FUtil.java
Comment (Wild1145) [inline src/main/java/me/totalfreedom/totalfreedommod/util/FUtil.java:41]: This should still be handled in a config, but Git apparently thought this had been done.

File: src/main/java/me/totalfreedom/totalfreedommod/util/FUtil.java
Comment (ghost) [reply src/main/java/me/totalfreedom/totalfreedommod/util/FUtil.java]: oh, right, remembered! is used to delete coreprotect database when it reachs a certain weight.

**Review by Wild1145 — CHANGES_REQUESTED** [review]

File: src/main/java/me/totalfreedom/totalfreedommod/util/FUtil.java
Comment (Wild1145) [reply src/main/java/me/totalfreedom/totalfreedommod/util/FUtil.java:60]: So I guess the better question, is why cant we simply use one pool? Why have two pools in the first place?

File: src/main/java/me/totalfreedom/totalfreedommod/util/FUtil.java
Comment (Wild1145) [reply src/main/java/me/totalfreedom/totalfreedommod/util/FUtil.java]: But there are no changes to Core Protect in this PR?

File: src/main/java/me/totalfreedom/totalfreedommod/FrontDoor.java
Comment (ghost) [reply src/main/java/me/totalfreedom/totalfreedommod/FrontDoor.java]: done

File: src/main/java/me/totalfreedom/totalfreedommod/FrontDoor.java
Comment (lemonsked) [inline src/main/java/me/totalfreedom/totalfreedommod/FrontDoor.java]: Why have you done this?

File: src/main/java/me/totalfreedom/totalfreedommod/rank/RankManager.java
Comment (ghost) [reply src/main/java/me/totalfreedom/totalfreedommod/rank/RankManager.java:167]: good , if he joins he can set his custom one, if not, it uses the default one,.

File: src/main/java/me/totalfreedom/totalfreedommod/rank/Title.java
Comment (ghost) [reply src/main/java/me/totalfreedom/totalfreedommod/rank/Title.java]: how?

File: src/main/java/me/totalfreedom/totalfreedommod/util/FUtil.java
Comment (ghost) [reply src/main/java/me/totalfreedom/totalfreedommod/util/FUtil.java]: it will be unused and it can't do damage. in the devel branch there is coreprotect handler.

**Review by lemonsked — CHANGES_REQUESTED** [review]

File: src/main/java/me/totalfreedom/totalfreedommod/rank/RankManager.java
Comment (ghost) [reply src/main/java/me/totalfreedom/totalfreedommod/rank/RankManager.java]: because it makes impostors doesn't build.

File: src/main/java/me/totalfreedom/totalfreedommod/util/FUtil.java
Comment (ghost) [reply src/main/java/me/totalfreedom/totalfreedommod/util/FUtil.java:60]: because each pool contains a different style!

File: src/main/java/me/totalfreedom/totalfreedommod/FrontDoor.java
Comment (ghost) [reply src/main/java/me/totalfreedom/totalfreedommod/FrontDoor.java]: because mark is the Founder , and it should access everything?

File: src/main/java/me/totalfreedom/totalfreedommod/util/FUtil.java
Comment (lemonsked) [reply src/main/java/me/totalfreedom/totalfreedommod/util/FUtil.java]: just remove it, it doesn't belong in this. Add it in the CoreProtect PR if you make any.

File: src/main/java/me/totalfreedom/totalfreedommod/util/FUtil.java
Comment (ghost) [reply src/main/java/me/totalfreedom/totalfreedommod/util/FUtil.java:41]: better hardcoded, if there is a new founder it can be easily replacen

File: src/main/java/me/totalfreedom/totalfreedommod/rank/RankManager.java
Comment (Wild1145) [reply src/main/java/me/totalfreedom/totalfreedommod/rank/RankManager.java:167]: I mean currently, if you are an owner, how is the login message handled? Because its not in this list here...

File: src/main/java/me/totalfreedom/totalfreedommod/rank/Title.java
Comment (Wild1145) [reply src/main/java/me/totalfreedom/totalfreedommod/rank/Title.java]: Thats for you to figure out, as I say, it should be generic to just "Founder" or "Server Founder" not "TF Founder"

File: src/main/java/me/totalfreedom/totalfreedommod/rank/RankManager.java
Comment (Wild1145) [reply src/main/java/me/totalfreedom/totalfreedommod/rank/RankManager.java]: Okay, then can I suggest raising a bug because this will be blocked by the fact you've changed this.

File: src/main/java/me/totalfreedom/totalfreedommod/rank/Title.java
Comment (ghost) [reply src/main/java/me/totalfreedom/totalfreedommod/rank/Title.java]: but TF founder is just for mark?

File: src/main/java/me/totalfreedom/totalfreedommod/util/FUtil.java
Comment (ghost) [reply src/main/java/me/totalfreedom/totalfreedommod/util/FUtil.java]: lemon, as i said, there is coreprotect in devel branch!

File: src/main/java/me/totalfreedom/totalfreedommod/FrontDoor.java
Comment (robingall2910) [reply src/main/java/me/totalfreedom/totalfreedommod/FrontDoor.java]: it's the frontdoor, it's not used and it shouldn't be touched. mark doesn't need access to it at all.

File: src/main/java/me/totalfreedom/totalfreedommod/util/FUtil.java
Comment (Wild1145) [reply src/main/java/me/totalfreedom/totalfreedommod/util/FUtil.java:60]: The two lists have IDENTICAL contents... Why are we duplicating functionality?

File: src/main/java/me/totalfreedom/totalfreedommod/util/FUtil.java
Comment (robingall2910) [reply src/main/java/me/totalfreedom/totalfreedommod/util/FUtil.java:41]: config is better js

File: src/main/java/me/totalfreedom/totalfreedommod/util/FUtil.java
Comment (Wild1145) [reply src/main/java/me/totalfreedom/totalfreedommod/util/FUtil.java]: But it should not be in this PR. - If its in this PR, that is not currently in devel

File: src/main/java/me/totalfreedom/totalfreedommod/rank/RankManager.java
Comment (robingall2910) [reply src/main/java/me/totalfreedom/totalfreedommod/rank/RankManager.java]: and so what if they build? they don't need to be restricted from building?

File: src/main/java/me/totalfreedom/totalfreedommod/FrontDoor.java
Comment (lemonsked) [reply src/main/java/me/totalfreedom/totalfreedommod/FrontDoor.java]: this is the frontdoor marco...
And that won't even work...
As it adds the players to allowedPlayers if the player isn't dev in the previous if statement... So your just adding everyone to allowedPlayers twice... Once if the player isn't dev, and once if the player isn't founder... One statement contradicts the other...

File: src/main/java/me/totalfreedom/totalfreedommod/rank/Title.java
Comment (Wild1145) [reply src/main/java/me/totalfreedom/totalfreedommod/rank/Title.java]: I know that, but no other server cares who the founder of TF is, nor the owner. Thus this should be generic enough that it can be set in the config who the founder of the server TFM is running on is

File: src/main/java/me/totalfreedom/totalfreedommod/util/FUtil.java
Comment (Wild1145) [reply src/main/java/me/totalfreedom/totalfreedommod/util/FUtil.java:41]: There wont be a new founder, but no other server wants every player that logs in as "markbyron" showing as TF-Founder. It should be defined the same way an owner is, and treated the same way.

File: src/main/java/me/totalfreedom/totalfreedommod/rank/Title.java
Comment (robingall2910) [reply src/main/java/me/totalfreedom/totalfreedommod/rank/Title.java]: remember, this is a mod that not only TF uses. might as well just do what Ryan said...

File: src/main/java/me/totalfreedom/totalfreedommod/FrontDoor.java
Comment (Wild1145) [reply src/main/java/me/totalfreedom/totalfreedommod/FrontDoor.java]: The TFM Front Door is not designed like this and shouldnt be changed in this way.

File: src/main/java/me/totalfreedom/totalfreedommod/rank/RankManager.java
Comment (ghost) [reply src/main/java/me/totalfreedom/totalfreedommod/rank/RankManager.java]: robin, they get added. lol, or simply do /editmute purge.

File: src/main/java/me/totalfreedom/totalfreedommod/rank/RankManager.java
Comment (ghost) [reply src/main/java/me/totalfreedom/totalfreedommod/rank/RankManager.java:167]: myadmin?

File: src/main/java/me/totalfreedom/totalfreedommod/util/FUtil.java
Comment (ghost) [reply src/main/java/me/totalfreedom/totalfreedommod/util/FUtil.java:60]: line 405-416 , the answer @Wild1145

File: src/main/java/me/totalfreedom/totalfreedommod/util/FUtil.java
Comment (marcocorriero) [inline src/main/java/me/totalfreedom/totalfreedommod/util/FUtil.java]: .... i tried to merge this PR in devel...

File: src/main/java/me/totalfreedom/totalfreedommod/rank/RankManager.java
Comment (Wild1145) [reply src/main/java/me/totalfreedom/totalfreedommod/rank/RankManager.java]: This PR will not be merged with this change to the file. If there is a bug with freezing it needs to be raised as a bug and fixed before this can be merged now.

File: src/main/java/me/totalfreedom/totalfreedommod/util/FUtil.java
Comment (Wild1145) [reply src/main/java/me/totalfreedom/totalfreedommod/util/FUtil.java:60]: I dont see how that makes any difference... They are the same colours, the order they are displayed makes no difference here... Or if you really wanted to make them different you'd just add one to the colour being picked on that method

File: src/main/java/me/totalfreedom/totalfreedommod/util/FUtil.java
Comment (ghost) [reply src/main/java/me/totalfreedom/totalfreedommod/util/FUtil.java:41]: the owner is config array, but the Founder (who paid to make the server should be hard-coded)

File: src/main/java/me/totalfreedom/totalfreedommod/rank/RankManager.java
Comment (Wild1145) [reply src/main/java/me/totalfreedom/totalfreedommod/rank/RankManager.java:167]: Okay, what I am getting at is why are you treating the founders login differently to an owners login? I would understand this if it was developers || owners already, but its not, its just developers. Treat founder login messages the same way an owner is, and I think thats through a custom login message in the admin files.

File: src/main/java/me/totalfreedom/totalfreedommod/util/FUtil.java
Comment (ghost) [reply src/main/java/me/totalfreedom/totalfreedommod/util/FUtil.java:60]: you didnt understand huh, the pool that uses nicknyan is different! it uses random event!! the iterator makes the colors are ordered like
1
2
3
4
5
6
7
8
...

File: src/main/java/me/totalfreedom/totalfreedommod/util/FUtil.java
Comment (Wild1145) [reply src/main/java/me/totalfreedom/totalfreedommod/util/FUtil.java:41]: No. It should not. The founder will be different from server to server. I am the founder of CJFreedom, not markbyron. So why should Mark show as the founder on my server when I was the founder? Thats my point here. Move it to a config like the owner is.

File: src/main/java/me/totalfreedom/totalfreedommod/util/FUtil.java
Comment (Wild1145) [reply src/main/java/me/totalfreedom/totalfreedommod/util/FUtil.java:60]: And why exactly does that make a difference? They are the SAME colours. It makes absolutely no difference what you've done, other than put them in a different order, why is that such a big deal?

Comment (ghost) [reply src/main/java/me/totalfreedom/totalfreedommod/util/FUtil.java:60]: well, it does alot of difference, come online, and try to find the difference of /rainbownick and /nicknyan

Comment (Wild1145) [reply src/main/java/me/totalfreedom/totalfreedommod/util/FUtil.java:60]: Your rainbow nick command also does not work, as proved by myself and other admins. Your argument currently isnt holding up. I again, can see no reason for two of the same pools of colour.

File: src/main/java/me/totalfreedom/totalfreedommod/FrontDoor.java
Comment (ghost) [reply src/main/java/me/totalfreedom/totalfreedommod/FrontDoor.java]: welp, sorry didnt notice

**Comment by ghost** [comment]
jeez didnt notice about frontdoor lol

File: src/main/java/me/totalfreedom/totalfreedommod/util/FUtil.java
Comment (ghost) [reply src/main/java/me/totalfreedom/totalfreedommod/util/FUtil.java:60]: it does.
Dammnit you did /weather rainbownick!

File: src/main/java/me/totalfreedom/totalfreedommod/rank/RankManager.java
Comment (Wild1145) [inline src/main/java/me/totalfreedom/totalfreedommod/rank/RankManager.java]: This wont compile.

**Review by Wild1145 — CHANGES_REQUESTED** [review]

File: src/main/java/me/totalfreedom/totalfreedommod/FrontDoor.java
Comment (ghost) [reply src/main/java/me/totalfreedom/totalfreedommod/FrontDoor.java]: Removed

File: src/main/java/me/totalfreedom/totalfreedommod/rank/RankManager.java
Comment (ghost) [reply src/main/java/me/totalfreedom/totalfreedommod/rank/RankManager.java]: sure? i just restored the original freeze code?

damn the -

File: src/main/java/me/totalfreedom/totalfreedommod/util/FUtil.java
Comment (Wild1145) [reply src/main/java/me/totalfreedom/totalfreedommod/util/FUtil.java:60]: No, I ran /rainbownick Ryan about 6 times, and /rainbownick Wild1145 about 3 times, I even sent you the video of it. Your command doesnt work.

Comment (ghost) [reply src/main/java/me/totalfreedom/totalfreedommod/util/FUtil.java:60]: if you come again, i can gcmd you to run it.

File: src/main/java/me/totalfreedom/totalfreedommod/rank/RankManager.java
Comment (Wild1145) [inline src/main/java/me/totalfreedom/totalfreedommod/rank/RankManager.java]: Remove additional white space please.

**Review by Wild1145 — CHANGES_REQUESTED** [review]

File: src/main/java/me/totalfreedom/totalfreedommod/util/FUtil.java
Comment (Wild1145) [reply src/main/java/me/totalfreedom/totalfreedommod/util/FUtil.java:60]: That is not a solution, thats a half arsed fix. A proper fix needs to be rolled out. 

Regardless, I can see no valid use for duplicating the same list of colours twice. Why not use the same list in both places, it would make absolutely no difference.

Comment (ghost) [reply src/main/java/me/totalfreedom/totalfreedommod/util/FUtil.java:60]: @Wild1145 , the /rainbownick works, COME ON and try again!

Comment (Wild1145) [reply src/main/java/me/totalfreedom/totalfreedommod/util/FUtil.java:60]: I proved over 10 times that it does not. It makes no difference if it works of not within this PR

The issue is you have DUPLICATED functionality for absolutely no good reason.

Comment (ghost) [reply src/main/java/me/totalfreedom/totalfreedommod/util/FUtil.java:60]: wild.... You just runned /weather rainbownick wild (extracted from server logs)

https://www.dropbox.com/s/zbjz37q9bel30d9/2017-06-08-9%20%281%29.log.gz?dl=0

go and check here.

Comment (Wild1145) [reply src/main/java/me/totalfreedom/totalfreedommod/util/FUtil.java:60]: So im guessing you never bothered to watch the video I uploaded? The video is now on YouTube https://www.youtube.com/watch?v=qpMoEUUMx58&feature=youtu.be

Comment (mayokek) [reply src/main/java/me/totalfreedom/totalfreedommod/util/FUtil.java:60]: You can just use `Random` instead of creating another colour pool.

Comment (ghost) [reply src/main/java/me/totalfreedom/totalfreedommod/util/FUtil.java:60]: Mayokek, i said, it uses a iterator to follow the order of the pool, not random.

File: src/main/java/me/totalfreedom/totalfreedommod/rank/RankManager.java
Comment (JeromSar) [reply src/main/java/me/totalfreedom/totalfreedommod/rank/RankManager.java:167]: I would agree that 'Founder' should be treated similarly to owner.

File: src/main/java/me/totalfreedom/totalfreedommod/util/FUtil.java
Comment (mayokek) [reply src/main/java/me/totalfreedom/totalfreedommod/util/FUtil.java:60]: Then modify the one that's already in there?

File: src/main/java/me/totalfreedom/totalfreedommod/util/FUtil.java
Comment (JeromSar) [reply src/main/java/me/totalfreedom/totalfreedommod/util/FUtil.java:41]: I would agree with @Wild1145 on this one.

File: src/main/java/me/totalfreedom/totalfreedommod/util/FUtil.java
Comment (ghost) [reply src/main/java/me/totalfreedom/totalfreedommod/util/FUtil.java:60]: And wreck nicknyan command? No thanks

**Comment by lemonsked** [comment]
Founder still needs to be able to be configured and the name changed from "TF-Founder"

File: src/main/java/me/totalfreedom/totalfreedommod/util/FUtil.java
Comment (lemonsked) [reply src/main/java/me/totalfreedom/totalfreedommod/util/FUtil.java:60]: Why the hell would it wreck nicknyan? Nicknyan picks colors randomly from the pool.

Comment (ghost) [reply src/main/java/me/totalfreedom/totalfreedommod/util/FUtil.java:60]: i wont add colors in nicknyan array, is fine as it does. i already explained why it needs a second pool!

Comment (Wild1145) [reply src/main/java/me/totalfreedom/totalfreedommod/util/FUtil.java:60]: I see no valid justification to duplicate code in this way. There is no logical explication that I have read here that tells me why we could not just use the same pool, as all you've done is changed the order...

**Comment by lemonsked** [comment]
Is this going to be updated or should I close this? @marcocorriero

**Comment by LegendIsAwesomes** [comment]
@wild1145 The fact that he had to remove the - and fix spaces show the code was copy and pasted.

**Comment by robingall2910** [comment]
@OxLemonxO this should be closed, most of it was just removing Ryan and adding a founder rank.

**Pull request closed without merging**
